File: RPC/nov-2023/Dictionary.cpp
#include <bits/stdc++.h>
using namespace std;

int buscar(string p, vector<string> word){
    int inf=0, sup=word.size();
    int m=(inf+sup)/2;
    while(inf+1<sup){
        if(word[m]>p) sup=m;
        else inf=m;
        m=(inf+sup)/2;
    }
    return inf;
}

int main(int argc, char const *argv[])
{
    ios_base::sync_with_stdio(false); cin.tie(0);
    int n, q;
    cin>>n>>q;
    vector<string> pal(n);
    for (int i = 0; i < n; i++)
    {
        cin>>pal[i];
    }
    sort(pal.begin(), pal.end());
    string a, b;
    int p, k;
    while(q--){
        cin>>a>>b;
        p=buscar(a, pal);
        k=buscar(b, pal);
        if(pal[p]==a || !p) cout<<k-p+1<<'\n';
        else cout<<k-p<<'\n';
    }
    return 0;
}

File: RPC/nov-2023/Dictionary.py
n, q=map(int, input().split())

dict=[]

for i in range(n):
    s=str(input())
    dict.append(s)

dict=sorted(dict)
mapa ={}

def buscar(pal):
    inf=0
    sup=n
    m=(inf+sup)//2
    while inf+1<sup:
        if dict[m]>pal: sup=m
        else: inf=m
        m=(inf+sup)//2
    return inf



while q:
    a, b=map(str, input().split())
    if a in mapa: p=mapa[a]
    else: p=buscar(a)
    if b in mapa: k=mapa[b]
    else: k=buscar(b)
    if dict[p]==a or p==0:  print(k-p+1)
    else: print(k-p)
    q-=1
